src/borg/testsuite/helpers.py

```diff
@@ -1133,3 +1134,15 @@ def test_iter_separated():
 def test_eval_escapes():
     assert eval_escapes('\\n\\0\\x23') == '\n\0#'
     assert eval_escapes('äç\\n') == 'äç\n'
+
+
+def test_safe_unlink_is_safe(tmpdir):
+    contents = b"Hello, world\n"
+    victim = tmpdir / 'victim'
+    victim.write_binary(contents)
+    hard_link = tmpdir / 'hardlink'
+    hard_link.mklinkto(victim)
+
+    safe_unlink(hard_link)
+
+    assert victim.read_binary() == contents
```

Comment:
well, it is ok that we have a test, but we also must face that this only tests the usual behaviour (which is the rather not-so-complicated part of `safe_unlink` and basically just a call to `os.unlink`) and not the more complicated part of it.

Reply:
hmm, can you check whether mocking `os.unlink` to raise ENOSPC works?